### PR TITLE
Mega GAM partially done, code cleanup

### PR DESCRIPTION
**Pull request by chrisng93**
- Correctly sort/order active and inactive GAM campaigns
- Add Mega GAM list item on GAM page
- Type a lot of things
- Little bit of code cleanup

Note that there's a ton of todos still. Will do in followups

Large screen:
Screen Shot 2020-11-16 at 11 09 17 PM

Small screen:
Screen Shot 2020-11-16 at 11 12 02 PM

Ref https://trello.com/c/tZJj4Wkb/725-megagam-active-mega-gam-campaign-list-item

### CODE REVIEW

File: src/components/MerchantsPage/gam/MegaGamListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+const MegaGamListItem = ({ campaign }: Props) => {
+  const { t } = useTranslation();
+  return (
+    <Container>
+      <HeroImage heroImageUrl={campaign.gallery_image_urls[0]} />
+      <Content>
+        {/* TODO: Figure out if this is the correct header. If so, we need to translate. */}
+        <Header>Mega-GAM</Header>
+        {/* TODO: Don't hard code this. Figure out where we should get this data from. */}
+        <Subheader>December Campaign</Subheader>
```
Comment (chrisng93) [inline src/components/MerchantsPage/gam/MegaGamListItem.tsx:21]: should the campaign model have a name? should we get this from the project? unsure yet

File: src/components/MerchantsPage/gam/MegaGamListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        <Description>{campaign.description}</Description>
+        <DonationContainer>
+          {/* TODO: Add donation data UI. */}
+          <DonationContent>_DONATION_DATA_</DonationContent>
+          {campaign.active && (
+            // TODO: Open payment modal.
+            <Button className="button--filled" onClick={undefined}>
+              {t('gamHome.listItem.giftButton')}
```
Comment (aliceluo2048) [inline src/components/MerchantsPage/gam/MegaGamListItem.tsx:29]: fyi Figma designs have the button as plural Gift Meals

Comment (chrisng93) [reply src/components/MerchantsPage/gam/MegaGamListItem.tsx]: oh fair, updated this to be plural for the eng translation. would anyone happen to know if we need new copy for chinese?

**Review by aliceluo2048 — APPROVED** [review]

File: src/utilities/api/types.ts [hunk 3]
```diff
@@ -106,21 +106,33 @@ export type BrowsePageSeller = {
   logo_image_url: string;
 };
 
+type SellerDistributorPair = {
+  distributor_id: number;
+  distributor_image_url: string;
+  distributor_name: string;
+  seller_id: number;
```
Comment (aliceluo2048) [inline src/utilities/api/types.ts:113]: `seller_id` is a string

Comment (chrisng93) [reply src/utilities/api/types.ts:113]: updated

**Pull request merged**